db/dataset_eval.py

```diff
@@ -68,8 +68,9 @@ def evaluate_dataset(dataset_id, normalize, eval_location, filter_type=None):
             raise JobExistsException
 
         # Validate dataset contents
-        validate_dataset_contents(db.dataset.get(dataset_id))
-        return _create_job(connection, dataset_id, normalize, eval_location, filter_type)
+        ignore_rec = validate_dataset_contents(db.dataset.get(dataset_id))
+        job_id = _create_job(connection, dataset_id, normalize, eval_location, filter_type)
```

Comment:
This is going to create an evaluation job containing all of the MBIDs in the dataset, including the ones that we have ignored. We should make sure that when we create the dataset snapshot that we don't include these recording ids, otherwise the dataset evaluation will fail:
https://github.com/metabrainz/acousticbrainz-server/blob/aeeec4d02b689b843efb65d4f4c58dd40ec20972/db/dataset.py#L280

Reply:
I used the same `filter_dataset_contents` before creating a snap